### PR TITLE
feat(tssdk): expose GraphQL client

### LINKED ISSUE #7159: ✨ Expose GraphQL client in TypeScript SDK
### What are you trying to do?

Expose the GraphQL client directory from the `dag` variable

See https://discord.com/channels/707636530424053791/1230428619755753553

### Why is this important to you?

Allow greater flexibility and special use case like calling other module

### How are you currently working around this?

Current workaround from [this discord message](https://discord.com/channels/707636530424053791/1230428619755753553/1230733941254783047)

```ts
connect(
  async (client: Client) => {
    await client
      .moduleSource("github.com/shykes/daggerverse/hello")
      .asModule()
      .initialize()
      .serve();
    const res = await (
      await client["_ctx"].connection()
    ).request(gql`
      query {
        hello {
          hello
        }
      }
    `);
    console.log(res);
  },
  { LogOutput: process.stderr },
);
```

### PR DESCRIPTION
**Pull request by TomChv**
Fixes #7159

### CODE REVIEW

**Comment by TomChv** [comment]
@helderco Java tests are failing which is not related to our PR, appart from that, everything is good :D

File: cmd/codegen/generator/typescript/templates/src/object.ts.gtpl [hunk 1]
```diff
@@ -47,6 +47,17 @@ export class {{ .Name | QueryToClient | FormatName }} extends BaseClient {
...
 
+      {{- /* Add custom method to main Client */ -}}
+      {{- if .Name | QueryToClient | FormatName | eq "Client" }}
+
+  /**
+   * Get the Raw GraphQL client.
+   */
+  public async getGQLClient() {
+    return this._ctx.connection()
+  }
```
Comment (helderco) [inline cmd/codegen/generator/typescript/templates/src/object.ts.gtpl:58]: This doesn't need to be async, does it?

**Review by helderco — APPROVED** [review]
It would be better to not have this be async. You'd need to make this simpler, yes, without trying to manage a lazy connection.

**Comment by TomChv** [comment]
I can make it possible easily

**Pull request merged**